packages/@react-spectrum/s2/src/TabsPicker.tsx

```diff
@@ -0,0 +1,431 @@
+  let domRef = useFocusableRef(ref);
+  props = useFormProps(props);
+  let {
+    direction = 'bottom',
+    align = 'start',
+    shouldFlip = true,
+    menuWidth,
+    children,
+    items,
+    size = 'M',
+    placeholder = stringFormatter.format('picker.placeholder'),
+    isQuiet,
+    density,
+    ...pickerProps
+  } = props;
+
+  // Better way to encode this into a style? need to account for flipping
+  let menuOffset: number;
+  if (size === 'S') {
+    menuOffset = 6;
+  } else if (size === 'M') {
+    menuOffset = 6;
+  } else if (size === 'L') {
+    menuOffset = 7;
+  } else {
+    menuOffset = 8;
+  }
+
+  return (
+    <AriaSelect
+      {...pickerProps}
+      placeholder={placeholder}>
+      {({isDisabled, isOpen, isInvalid}) => (
+        <>
+          <InternalPickerContext.Provider value={{size}}>
+            <FieldLabel isQuiet={isQuiet} />
+            <Button
+              ref={domRef}
+              style={renderProps => pressScale(domRef)(renderProps)}
+              // Prevent press scale from sticking while Picker is open.
+              // @ts-ignore
+              isPressed={false}
+              className={renderProps => inputButton({
+                ...renderProps,
+                size: size,
+                isOpen,
+                isQuiet,
+                density
+              })}>
+              {(renderProps) => (
+                <>
+                  <SelectValue className={valueStyles({isQuiet}) + ' ' + raw('&> * {display: none;}')}>
+                    {({defaultChildren}) => {
+                      return (
+                        <Provider
+                          values={[
+                            [IconContext, {
+                              slots: {
+                                icon: {
+                                  render: centerBaseline({slot: 'icon', styles: iconCenterWrapper}),
+                                  styles: icon
+                                }
+                              }
+                            }],
+                            [TextContext, {
+                              slots: {
+                                // Default slot is useful when converting other collections to PickerItems.
+                                [DEFAULT_SLOT]: {styles: style({
+                                  display: 'block',
+                                  flexGrow: 1,
+                                  truncate: true
+                                })}
+                              }
+                            }],
+                            [InsideSelectValueContext, true]
+                          ]}>
+                          {defaultChildren}
+                        </Provider>
+                      );
+                    }}
+                  </SelectValue>
+                  <ChevronIcon
+                    size={size}
+                    className={iconStyles} />
+                  {isInvalid && !isDisabled && !isQuiet &&
```

Comment:
can we have invalid pickers in a collapsed tabs or can we remove this?

Reply:
yeah, there's a bunch of styling on the TabsPicker that I haven't bothered to clean up yet
you are correct though, we can't have an invalid one